src/lval.js

```diff
@@ -172,6 +172,10 @@ pp.parseMaybeDefault = function(startPos, startLoc, left) {
 
 // Verify that a node is an lval — something that can be assigned
 // to.
+// bindingType can be either:
+// 'var' indicating that the lval creates a 'var' binding
+// 'let' indicating that the lval creates a lexical ('let' or 'const') binding
+// 'none' indicating that the binding should be checked for illegal identifiers, but not for duplicate references
```

Comment:
Eventually, I think it would be a good idea to handle binding tracking separately from `checkLVal`. That would require a bit of refactoring though, so this is intended to be a simpler patch for the current issue.